clap_complete/src/dynamic/completer.rs

```diff
@@ -298,9 +361,51 @@ fn complete_arg(
                 completions.extend(complete_arg_value(arg.to_value(), positional, current_dir));
             }
         }
-        ParseState::Opt(opt) => {
-            completions.extend(complete_arg_value(arg.to_value(), &opt, current_dir));
-        }
+        ParseState::Opt(opt) => match opt.get_value_delimiter() {
+            Some(delim) => {
+                let values = {
+                    let mut delim_buffer = [0; 4];
+                    let delim = delim.encode_utf8(&mut delim_buffer);
+
+                    arg.to_value_os()
+                        .split(delim)
+                        .into_iter()
+                        .collect::<Vec<_>>()
+                };
+                let last = values.last();
```

Comment:
Every place we can complete a value, we shouldn't need to support delimiters.

Is there a reason this isn't living inside of `complete_arg_value`?

Reply:
Oh my gosh, I messed up. It was right to put this in complete_arg_value, and it will naturally support the positional delimiter value.